### PR TITLE
update client lib to api spec 2.2.0

### LINKED ISSUE #50: update client lib to 2.2.0 rest spec
A few small model changes have occured, the models need to be updated.
https://github.com/Senzing/senzing-rest-api-specification/releases/tag/2.2.0

- Added `SzNameScoring` to describe name scoring details
- Added `SzSearchFeatureScore` for search feature scores
- Modified `SzBaseRelatedEntity` to remove `fullNameScore` field since it has
not been populated since switch to version 2.0.0 of native Senzing SDK and
never made sense in the "base class" since only `SzAttributeSearchResult` had
this field populated under native Senzing SDK version 1.x.
- Added `bestNameScore` field to `SzAttributeSearchResult` to replace
`fullNameScore` in the place where the name score was previously
used with version 1.x of the native Senzing SDK (i.e.: to sort search results
based on the strength of the name match).
- Modified `SzAttributeSearchResult` to include the `featureScores` field to
provide feature scores without using "raw data"
- Added `nameScoringDetails` field to `SzFeatureScore` class to provide
`SzNameScoring` name scoring details on why operations,
- Updated `com.senzing.api.model.SzFeatureScore` to define its `score` field as 
the most sensible score value from the `nameScoringDetails` for `"NAME"`
features since the `FULL_SCORE` field is not available for names.
- Added the `NOT_SCORED` value for `SzScoringBucket` enum.

### PR DESCRIPTION
**Pull request by arawinters**
# Pull request questions

## Which issue does this address

resolves #50 

## Why was change needed

to match api spec 2.2.0 https://github.com/Senzing/senzing-rest-api-specification/releases/tag/2.2.0

### CODE REVIEW

File: projects/rest-api-client-ng/src/lib/model/szFeatureScore.ts [hunk 1]
```diff
@@ -2,31 +2,33 @@
...
+import { SzNameScoring } from './szNameScoring';
 import { SzScoredFeature } from './szScoredFeature';
 import { SzScoringBehavior } from './szScoringBehavior';
 import { SzScoringBucket } from './szScoringBucket';
 
 /**
  * Describes the scoring between two `SzScoredFeature` instances.
  */
-export interface SzFeatureScore {
+export interface SzFeatureScore {
```
Comment (docktermj) [inline projects/rest-api-client-ng/src/lib/model/szFeatureScore.ts:20]: A little nit here with a trailing space.  I've seen it in other places, too.

File: projects/rest-api-client-ng/src/lib/model/szMatchInfo.ts [hunk 2]
```diff
@@ -2,24 +2,26 @@
...
  */
 import { SzCandidateKey } from './szCandidateKey';
 import { SzFeatureScore } from './szFeatureScore';
+import { SzMatchLevel } from './szMatchLevel';
 
 /**
  * The match info describing why two entities (or records) resolve or relate to one another.
  */
-export interface SzMatchInfo {
+export interface SzMatchInfo {
```
Comment (docktermj) [inline projects/rest-api-client-ng/src/lib/model/szMatchInfo.ts:19]: Like here.  (Trailing space)

File: projects/rest-api-client-ng/src/lib/model/szRelatedEntity.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ import { SzRelationshipType } from './szRelationshipType';
 /**
  * Provides a description of an entity that is related to a ResolvedEntity.  This describes how the entity is related and may be missing the complete features and record list of a ResolvedEntity.
  */
-export interface SzRelatedEntity extends SzBaseRelatedEntity {
+export interface SzRelatedEntity extends SzBaseRelatedEntity {
```
Comment (docktermj) [inline projects/rest-api-client-ng/src/lib/model/szRelatedEntity.ts:18]: And here.  (There are more, but won't comment on them.)

**Review by docktermj — APPROVED** [review]

File: projects/rest-api-client-ng/src/lib/model/szFeatureScore.ts (hunk 1 above)
Comment (arawinters) [reply projects/rest-api-client-ng/src/lib/model/szFeatureScore.ts:20]: generator. I have two choices:
1) fix it, and then on next update have to deal with the same thing again.
2) leave it, and not deal have to have git flag it as a change.

**Pull request merged**